homeassistant/components/natural_resources_wales/sensor.py

```diff
@@ -0,0 +1,50 @@
+"""Platform for natural resources wales sensor."""
+from datetime import timedelta
+import logging
+
+import voluptuous as vol
+
+import homeassistant.components.natural_resources_wales.river_levels_sensor as rls
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import CONF_API_KEY, CONF_NAME, CONF_SCAN_INTERVAL
+import homeassistant.helpers.config_validation as cv
+
+SCAN_INTERVAL = timedelta(seconds=60)
+DEFAULT_NAME = "Natural Resources Wales"
+DEFAULT_LANGUAGE = "en"
+CONF_LANGUAGE = "language"
+LANGUAGE_CODES = [
+    "en",
+    "cy",
+]
+CONF_MONITORED_STATIONS = "monitored_stations"
+
+
+_LOGGER = logging.getLogger(__name__)
+
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
```

Comment:
We don't allow for configuration via a platform integration anymore and require configuring an integration via a configuration flow.


For more information, check out ADR-0007 & ADR-0010:

- https://github.com/home-assistant/architecture/blob/master/adr/0007-integration-config-yaml-structure.md
- https://github.com/home-assistant/architecture/blob/master/adr/0010-integration-configuration.md

For the developer documentation on building a configuration flow see:

https://developers.home-assistant.io/docs/config_entries_config_flow_handler